webapp/channels/src/components/admin_console/access_control/policy_details/policy_details.tsx

```diff
@@ -0,0 +1,412 @@
+    actions: PolicyActions;
+    autocompleteResult: AccessControlExpressionAutocomplete;
+}
+
+interface ChannelChanges {
+    removed: Record<string, ChannelWithTeamData>;
+    added: Record<string, ChannelWithTeamData>;
+    removedCount: number;
+}
+
+const PolicyDetails: React.FC<PolicyDetailsProps> = ({policyId, actions}) => {
+    const [policyName, setPolicyName] = useState('');
+    const [expression, setExpression] = useState('');
+    const [autoSyncMembership, setAutoSyncMembership] = useState(false);
+    const [serverError, setServerError] = useState(false);
+    const [addChannelOpen, setAddChannelOpen] = useState(false);
+    const [editorMode, setEditorMode] = useState<'cel' | 'table'>('cel');
+    const [channelChanges, setChannelChanges] = useState<ChannelChanges>({
+        removed: {},
+        added: {},
+        removedCount: 0,
+    });
+    const [saveNeeded, setSaveNeeded] = useState(false);
+    const [channelsCount, setChannelsCount] = useState(0);
+    const [autocompleteResult, setAutocompleteResult] = useState<AccessControlExpressionAutocomplete>({entities: {}});
+
+    useEffect(() => {
+        loadPage();
+    }, [policyId]);
+
+    const loadPage = async () => {
+        if (policyId) {
+            const result = await actions.fetchPolicy(policyId);
+
+            // Set policy name and expression after fetching the policy
+            setPolicyName(result.data?.name || '');
+            setExpression(result.data?.rules?.[0]?.expression || '');
+
+            // Search for channels after setting the policy details
+            const channelsResult = await actions.searchChannels(policyId, '', {per_page: DEFAULT_PAGE_SIZE});
+            setChannelsCount(channelsResult.data?.total_count || 0);
+
+            const autocompleteResult = await actions.getAccessControlExpressionAutocomplete();
+            if (autocompleteResult.data) {
+                setAutocompleteResult(autocompleteResult.data);
+            }
+        }
+    };
+
+    const handleSubmit = async () => {
+        try {
+            const updatedPolicy = await actions.createPolicy({
+                id: policyId || '',
+                name: policyName,
+                rules: [{expression, actions: []}] as AccessControlPolicyRule[],
+                type: 'parent',
+                version: 'v0.1',
+            });
+
+            if (updatedPolicy.data.id) {
+                if (channelChanges.removedCount > 0) {
+                    await actions.unassignChannelsFromAccessControlPolicy(updatedPolicy.data.id, Object.keys(channelChanges.removed));
+                }
+                if (Object.keys(channelChanges.added).length > 0) {
+                    await actions.assignChannelsToAccessControlPolicy(updatedPolicy.data.id, Object.keys(channelChanges.added));
+                }
+            }
+
+            getHistory().push('/admin_console/user_management/attribute_based_access_control/edit_policy/' + updatedPolicy.data.id);
```

Comment:
I think we should use `encodeURI` when appending variables into URL for security purposes.